### DISCUSSION TITLE
Build images with BentoML base image

### DISCUSSION THREAD

**Discussion by mat-berna** [containers/buildah#3090] [Q&A]
Hi,

I'm trying to build a container image from BentoML base image (https://docs.bentoml.org/en/latest/quickstart.html#containerize-model-api-server) and push it on Docker Hub.
I'm using Buildah technology inside a Tekton task manifest:

> name: 3-build-buildah
> workingdir: /workspace
> image: quay.io/buildah/stable:latest
> script: |
>   #!/usr/bin/env bash
>   sed -i -e 's/=\ \"overlay\"/=\ \"vfs\"/g' /etc/containers/storage.conf
>   echo "------------------------------ Find bentoml bundle"
>   ...
>   echo "------------------------------ Start building bentoml bundle"
>   cd $LAST_IMAGE_PATH
>   ls -lh
>   buildah bud --layers --file $LAST_IMAGE_PATH/Dockerfile --tag $CR_REPO:$LAST_IMAGE --tag $CR_REPO:latest .
>   echo "------------------------------ Start pushing"
>   buildah push --tls-verify false $CR_REPO:$LAST_IMAGE
> volumeMounts:
>   name: build-task-volume
>   mountPath: /workspace
>   name: build-container-registry-volume
>   mountPath: "/root/.docker"
> securityContext:  
>   privileged: true
>   runAsUser: 0

When I execute `buildah bud` command I have the following error:

> STEP 4: COPY environment.yml requirements.txt setup.sh* bentoml-init.sh python_version* /bento/
> error building at STEP 
> "COPY environment.yml requirements.txt setup.sh* bentoml-init.sh python_version* /bento/": 
> error adding sources 
> [
>     /tekton/home/bentoml/repository/IrisClassifier/20210314110109_58D216/environment.yml 
>     /tekton/home/bentoml/repository/IrisClassifier/20210314110109_58D216/requirements.txt 
>     /tekton/home/bentoml/repository/IrisClassifier/20210314110109_58D216/setup.sh* 
>     /tekton/home/bentoml/repository/IrisClassifier/20210314110109_58D216/bentoml-init.sh 
>     /tekton/home/bentoml/repository/IrisClassifier/20210314110109_58D216/python_version*
> ]
> error checking on source /tekton/home/bentoml/repository/IrisClassifier/20210314110109_58D216/setup.sh* 
> under "/tekton/home/bentoml/repository/IrisClassifier/20210314110109_58D216": copier: stat: "/setup.sh*": no such file or directory 

I tried with similar technologies that allows to build a container image inside another container, e.g. Kaniko and DIND (Docker in Docker).
With these technologies I haven't errors.

Can you help me?
Here you can find the source code to build container image with Dockerfile: [irisclassifier-20210319171817-c36331.tar.gz](https://github.com/containers/buildah/files/6177776/irisclassifier-20210319171817-c36331.tar.gz)

Thanks.

**Comment by rhatdan** [comment]
I could only get this to happen if setup.sh did not exists.  Does it not exists in your tests?  Does Docker and Kaniko allow specification with * that do not exist?

**Reply by mat-berna** [reply]
Exactly.
The files setup/* don't exist, however Docker and Kaniko don't give errors.
The Dockerfile is auto generated from BentoML and maybe in some contexts these file are necessary.
